### ISSUE TITLE
Incomplete py-cython build 

### ISSUE THREAD

**Issue by ikitayama** [LLNL/spack#18554]
`py-cython` build seems to be broken at least on ThunderX2.

```
$ cython --version
Traceback (most recent call last):
  File "/home/users/itaru.kitayama/projects/spack/opt/spack/linux-centos8-thunderx2/gcc-8.3.1/py-cython-0.29.21-3637bpwma3qsf4oxxehkna62ysmz7tqv/bin/cython", line 10, in <module>
    from importlib.metadata import distribution
ModuleNotFoundError: No module named 'importlib.metadata'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/users/itaru.kitayama/projects/spack/opt/spack/linux-centos8-thunderx2/gcc-8.3.1/py-cython-0.29.21-3637bpwma3qsf4oxxehkna62ysmz7tqv/bin/cython", line 13, in <module>
    from importlib_metadata import distribution
ModuleNotFoundError: No module named 'importlib_metadata'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/users/itaru.kitayama/projects/spack/opt/spack/linux-centos8-thunderx2/gcc-8.3.1/py-cython-0.29.21-3637bpwma3qsf4oxxehkna62ysmz7tqv/bin/cython", line 15, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3095, in <module>
    @_call_aside
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3079, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3108, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 570, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 888, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 774, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'Cython==0.29.21' distribution was not found and is required by the application
```

**Comment by ikitayama** [comment]
This is resolved.

**Comment by scheibelp** [comment]
Hello @ikitayama, was this resolved by https://github.com/spack/spack/pull/18572? It looked like there was still a problem there (at https://github.com/spack/spack/pull/18572#issuecomment-689310500).

Or if you fixed it yourself could you mention how you did that?

**Comment by ikitayama** [comment]
I closed because the issue can be discussed in #18572.
